doc/whatsnew/2.14.rst

```diff
@@ -38,6 +38,10 @@ Extensions
 Other Changes
 =============
 
+* Exclude dunder calls on super() from raising ``unnecessary-dunder-call``.
+
+  Closes #6074
+
```

Comment:
I think the what's new can be skipped because we got it in before 2.14. (By the way, thanks for the fast follow-up!)
```suggestion
```